src/Sulu/Bundle/SecurityBundle/Resources/js/containers/Permissions/Permissions.js

```diff
@@ -0,0 +1,224 @@
+// @flow
+import React, {Fragment} from 'react';
+import {action, autorun, computed, observable} from 'mobx';
+import {observer} from 'mobx-react';
+import {Loader} from 'sulu-admin-bundle/components';
+import {MultiSelect} from 'sulu-admin-bundle/containers';
+import securityContextsStore from '../../stores/SecurityContextsStore';
+import type {SecurityContextGroups, SecurityContexts} from '../../stores/SecurityContextsStore/types';
+import type {ContextPermission} from './types';
+import permissionsStyle from './permissions.scss';
+import PermissionMatrix from './PermissionMatrix';
+
+type Props = {
+    system: string,
+    onChange: (value: Array<ContextPermission>) => void,
+    value: Array<ContextPermission>,
+};
+
+@observer
+export default class Permissions extends React.Component<Props> {
+    static webspacePlaceholder = '#webspace#';
+
+    systemDisposer: () => void;
+
+    @observable securityContextGroups: SecurityContextGroups;
+
+    @action componentDidMount() {
+        this.systemDisposer = autorun(
+            () => {
+                securityContextsStore.loadSecurityContextGroups(this.system).then(action((securityContextGroups) => {
+                    this.securityContextGroups = securityContextGroups;
+                }));
+            }
+        );
+    }
+
+    componentWillUnmount() {
+        this.systemDisposer();
+    }
+
+    @computed get system(): string {
+        return this.props.system;
+    }
+
+    @computed get webspaceContextPermissionPrefix(): string {
+        if (this.webspaceSecurityContextGroupKey) {
+            const securityContextGroup = this.securityContextGroups[this.webspaceSecurityContextGroupKey];
+            for (const securityContextKey of Object.keys(securityContextGroup)) {
+                if (securityContextKey.indexOf(Permissions.webspacePlaceholder) !== -1) {
```

Comment:
I think most of the `array.indexOf(value) === -1` calls in this file would be a bit easier to read if they use `array.includes(value)`, since the intention is clearer then (at some points it is hard to distinguish if you really need the index or it is just about finding if subset is there). This also works for strings.